test/rpc_test.py

```diff
@@ -1319,6 +1319,26 @@ def test_debug_info(self):
         expected.update(autograd_info)
         self.assertEqual(expected, info)
 
+    @requires_process_group_agent("Testing exceptions thrown by ProcessGroupAgent::enqueueSend.")
+    def test_sender_exceptions(self):
+        rpc.init_rpc(
+            name="worker%d" % self.rank,
+            backend=rpc.backend_registry.BackendType[
+                dist_utils.TEST_CONFIG.rpc_backend_name
+            ],
+            rank=self.rank,
+            world_size=self.world_size,
+            rpc_backend_options=self.rpc_backend_options,
+        )
+
+        if self.rank == 0:
+            time.sleep(3)
```

Comment:
I added a method `wait_until_node_failure` in dist_autograd_test.py. You can refactor that to dist_utils and reuse it here instead of time.sleep.